dispatcher/main.go

```diff
@@ -0,0 +1,480 @@
+package main
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/base32"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/Azure/adx-automation-agent/common"
+	"github.com/Azure/adx-automation-agent/httputils"
+	"github.com/Azure/adx-automation-agent/kubeutils"
+	"github.com/Azure/adx-automation-agent/models"
+)
+
+var (
+	indexScript   = getFilepath("get_index")
+	metadataYml   = getFilepath("metadata.yml")
+	httpClient    = &http.Client{CheckRedirect: nil}
+	namespace     = getNamespace()
+	droidMetadata = getDroidMetadata()
+	clientset     = kubeutils.TryCreateKubeClientset()
+)
+
+func main() {
+	info("A01 Dispatcher start running ...")
+	info(fmt.Sprintf("Pod name: %s", os.Getenv(common.EnvPodName)))
+
+	var pRunID *int
+	pRunID = flag.Int("run", -1, "The run ID")
+	flag.Parse()
+
+	if *pRunID == -1 {
+		log.Fatal("Missing runID")
+	}
+
+	run := getRun(*pRunID)
+
+	err := postTasks(run, queryTests(run))
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	jobDef, err := createTaskJob(run)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	// Ignore this error for now. This API's latest version seems to sending inaccurate
+	// error
```

Comment:
Out of curiosity, what is it sending?

Reply:
It says the `controller-uid` label is incorrect. However, it is actually correct and the job runs just fine.